### PR TITLE
Make the new autocomplete endpoint OCS

### PR DESCRIPTION
**Pull request by nickvergessen**
The new endpoint is much nicer and should be available for apps, so e.g. the android/ios talk app can also autocomplete in the chat in the future....
(The old sharee OCS endpoint does not allow to filter for "participants in a room")

### CODE REVIEW

File: core/Controller/AutoCompleteController.php [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 namespace OC\Core\Controller;
 
-use OCP\AppFramework\Controller;
+use OCP\AppFramework\OCSController as Controller;
```
Comment (MorrisJobke) [inline core/Controller/AutoCompleteController.php]: This shows really well how good that code is regarding transport abstraction. Just changing the base class and you run OCS style 🚀

Comment (nickvergessen) [reply core/Controller/AutoCompleteController.php]: Yeah and the data format is changed `ocs->data->[earlier data]`

But yes, I was quite happy too

**Review by rullzer — APPROVED** [review]
Cool stuff!

File: apps/comments/js/commentstabview.js [hunk 2]
```diff
@@ -240,19 +240,22 @@
...
-					OC.generateUrl('/autocomplete/get'),
-					{
+				this._autoCompleteRequestCall = $.ajax({
+					url: OC.linkToOCS('core', 2) + 'autocomplete/get',
+					data: {
 						search: query,
 						itemType: 'files',
 						itemId: s.model.get('id'),
 						sorter: 'commenters|share-recipients',
 						limit: OC.appConfig.comments.maxAutoCompleteResults
```
Comment (MorrisJobke) [inline apps/comments/js/commentstabview.js]: This one is for me not provisioned :/ But this is not related to the PR here.

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍
